service/history/execution/context.go

```diff
@@ -499,6 +499,7 @@ func (c *contextImpl) ConflictResolveWorkflowExecution(
 		NewWorkflowSnapshot:     newWorkflow,
 		CurrentWorkflowMutation: currentWorkflow,
 		// Encoding, this is set by shard context
+		DomainName: c.GetDomainName(),
```

Comment:
Could you change it to be the same as this implementation? 
https://github.com/uber/cadence/blob/master/service/history/execution/context.go#L662

c.GetDomainName() can return on cache failure